components/web-components/select/index.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react'
+
+import * as S from './style'
+
+type ValueType = string | number
+
+interface SelectProps {
+  data: Array<{ value: ValueType, text: string }>
```

Comment:
How about keep it simple and just accept child elements, which would be `<option>` elements, just like regular <select`? This gives freedon to use `<option disabled ...` for descriptive elements, e.g. _Choose a flavor_, or `<optgroup` etc etc.

I think it's ok since it's named Select so implies it's definitely a browser control which displays a native list of options, and means we don't have to make a complicated data type where we allow grouping etc. Also <option> elements aren't like regular elements - they can't be styled so there's little risk there.

Reply:
makes sense